dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-raft/src/main/java/org/apache/dolphinscheduler/plugin/registry/raft/RaftRegistry.java

```diff
@@ -0,0 +1,228 @@
+        return readUtf8(kvStore.bGet(key));
+    }
+
+    @Override
+    public void put(String key, String value, boolean deleteOnDisconnect) {
+        kvStore.bPut(key, writeUtf8(value));
+        if (key.startsWith(Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS + Constants.SINGLE_SLASH)) {
+            addWorkerGroup(key);
+        }
+    }
+
+    private void addWorkerGroup(String key) {
+        List<String> workerGroupList = getWorkerGroups();
+        String workerGroup = key.substring(key.indexOf(Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS)
+                + Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS.length() + 1, key.lastIndexOf(Constants.SINGLE_SLASH));
+        if (!workerGroupList.contains(workerGroup)) {
+            workerGroupList.add(workerGroup);
+            kvStore.bPut(REGISTRY_DOLPHINSCHEDULER_WORKER_GROUPS, writeUtf8(JSONUtils.toJsonString(workerGroupList)));
+        }
+    }
+
+    private List<String> getWorkerGroups() {
+        final String storedWorkerGroup = readUtf8(kvStore.bGet(REGISTRY_DOLPHINSCHEDULER_WORKER_GROUPS));
+        if (StringUtils.isEmpty(storedWorkerGroup)) {
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(JSONUtils.toList(storedWorkerGroup, String.class));
+    }
+
+    @Override
+    public void delete(String key) {
+        kvStore.bDelete(key);
+        final DistributedLock<byte[]> distributedLock = distributedLockMap.get(key);
+        if (distributedLock != null) {
+            distributedLock.unlock();
+        }
+        distributedLockMap.remove(key);
+    }
+
+    @Override
+    public Collection<String> children(String key) {
+        List<String> children = new ArrayList<>();
+        if (key.equals(Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS)) {
+            //get all worker groups
+            children = getWorkerGroups();
+
+        } else {
+            final List<KVEntry> result = kvStore.bScan(key, key + Constants.SINGLE_SLASH + Constants.RANDOM_STRING);
+            if (result.isEmpty()) {
+                return new ArrayList<>();
+            }
+            for (final KVEntry kv : result) {
+                final String entryKey = readUtf8(kv.getKey());
+                if (StringUtils.isEmpty(readUtf8(kv.getValue())) || StringUtils.isEmpty(entryKey)) {
+                    continue;
+                }
+                String child = entryKey.substring(entryKey.lastIndexOf(Constants.SINGLE_SLASH) + 1);
+                children.add(child);
+            }
+        }
+        children.sort(Comparator.reverseOrder());
+        return children;
+    }
+
+    @Override
+    public boolean exists(String key) {
+        return kvStore.bContainsKey(key);
+    }
+
+    @Override
+    public boolean acquireLock(String key) {
+        final DistributedLock<byte[]> distributedLock = kvStore.getDistributedLock(key, properties.getDistributedLockTimeout().toMillis(), TimeUnit.MILLISECONDS);
+        final boolean lock = distributedLock.tryLock();
```

Comment:
Does this method will return false? In fact the `acquireLock` need to block until get lock.